website/addons/googledrive/model.py

```diff
@@ -111,218 +101,183 @@ def get_or_create(cls, node, path):
-            self.oauth_settings = None
-            self.save()
-
-        for node_settings in self.googledrivenodesettings__authorized:
-            node_settings.deauthorize(Auth(self.owner))
-            node_settings.save()
-
-    def save(self, *args, **kwargs):
-        if self.oauth_settings:
-            self.oauth_settings.save()
-        return super(GoogleDriveUserSettings, self).save(*args, **kwargs)
-
-    def delete(self, save=True):
-        self.clear()
-        super(GoogleDriveUserSettings, self).delete(save)
-
-    def __repr__(self):
-        return u'<GoogleDriveUserSettings(user={self.owner.username!r})>'.format(self=self)
-
-
-class GoogleDriveNodeSettings(StorageAddonBase, AddonNodeSettingsBase):
-
-    folder_id = fields.StringField(default=None)
-    folder_path = fields.StringField(default=None)
-
-    user_settings = fields.ForeignField(
-        'googledriveusersettings', backref='authorized'
-    )
+    def provider_name(self):
+        return 'googledrive'
 
-    @property
     def folder_name(self):
-        if not self.folder_id:
+        if not self.drive_folder_id:
             return None
 
         if self.folder_path != '/':
-            return unquote(os.path.split(self.folder_path)[1].encode('utf-8')).decode('utf-8')
+            return urllib.unquote(os.path.split(self.folder_path)[1].encode('utf-8')).decode('utf-8')
 
-        return '/ (Full Google Drive)'
+    def clear_auth(self):
+        self.drive_folder_id = None
+        self.folder_path = None
+        self.drive_folder_name = None
+        return super(GoogleDriveNodeSettings, self).clear_auth()
 
-    @property
-    def has_auth(self):
-        """Whether an access token is associated with this node."""
-        return bool(self.user_settings and self.user_settings.has_auth)
+    def set_auth(self, *args, **kwargs):
+        self.drive_folder_id = None
+        return super(GoogleDriveNodeSettings, self).set_auth(*args, **kwargs)
 
-    @property
-    def complete(self):
-        return self.has_auth and self.folder_id is not None
+    def set_target_folder(self, folder, auth):
+        """Configure this addon to point to a Google Drive folder
 
-    def deauthorize(self, auth=None, add_log=True):
-        """Remove user authorization from this node and log the event."""
-        if add_log:
-            extra = {'folder': self.folder_name}
-            nodelogger = GoogleDriveNodeLogger(node=self.owner, auth=auth)
-            nodelogger.log(action="node_deauthorized", extra=extra, save=True)
+        :param dict folder:
+        :param User user:
+        """
+        self.drive_folder_id = folder['id']
+        self.folder_path = folder['path']
+        self.drive_folder_name = folder['name']
 
-        self.folder_id = None
-        self.folder_path = None
-        self.user_settings = None
+        # Tell the user's addon settings that this node is connecting
+        self.user_settings.grant_oauth_access(
```

Comment:
Where is `user_settings` defined? The only implementation I can find for `grant_oauth_access` is in `AddonOAuthUserSettingsBase`, which this class inherits from.  (so why use self.user_settings.x instead of self.x?)